lib/core-client/src/preview/StoryRenderer.tsx

```diff
@@ -77,13 +79,74 @@ export class StoryRenderer {
       this.channel.on(Events.STORY_ARGS_UPDATED, () => this.forceReRender());
       this.channel.on(Events.GLOBALS_UPDATED, () => this.forceReRender());
       this.channel.on(Events.FORCE_RE_RENDER, () => this.forceReRender());
+      this.channel.on(Events.DOCS_TARGETTED_RENDER, (data) => {
+        setTimeout(() => {
+          this.renderDocsStoryOutOfContext(data);
+        }, 1);
+      });
     }
   }
 
   forceReRender() {
     this.renderCurrentStory(true);
   }
 
+  async renderDocsStoryOutOfContext({ id, identifier, name }: Record<string, string>) {
+    // TODO, the channel seems to invoke this twice
+    if (cache[id]) {
+      return;
+    }
+    cache[id] = true;
+
+    const targetDOMNode = document.getElementById(identifier);
+
+    if (!targetDOMNode) {
+      cache[id] = false;
+      return;
+    }
+
+    const {
+      metadata,
+      context,
+    }: {
+      metadata: RenderMetadata;
+      context: RenderContextWithoutStoryContext;
+    } = this.getStoryMetadataAndContext(this.storyStore, id, 'docs', false);
+
+    const { getDecorated } = metadata;
+
+    if (getDecorated) {
+      try {
+        const { applyLoaders, unboundStoryFn } = context;
+        console.time(`applyLoaders_${id}`);
+        const storyContext = await applyLoaders();
+        console.timeEnd(`applyLoaders_${id}`);
+        const storyFn = () => unboundStoryFn(storyContext);
+
+        if (targetDOMNode.querySelector('[data-is-loadering-indicator="true"')) {
+          targetDOMNode.querySelector('[data-is-loadering-indicator="true"').remove();
```

Comment:
for angular it forces the full rendering every time. 🤔 
which differs from the canvas rendering which is optimized when only the args change